### PR TITLE
Fixed sending force to refresh/mode/set causes the gateway to refresh indefinitely

### PR DESCRIPTION
**Pull request by tisoft**

### CODE REVIEW

**Comment by tisoft** [comment]
@mschmiedel This fix just prevents setting previousMode to the same value as currentMode. We should really think about setting force mode in the handleMQTTCommand only if it was a real command for the car, not some configuration change for the gateway. As it is currently sending off, will always do a force at the same time, which is probably not what you want, when explicitly sending off. :)

But since you are doing some refactoring anyway, I didn't want to change the handling there for now.

**Comment by mschmiedel** [comment]
> But since you are doing some refactoring anyway, I didn't want to change the handling there for now.

I would go this way directly to be honest, trying to catch this in the VehicleState seems a bit risky to me. Just let us move the setRefreshMode(FORCE) to every switch case except for the refresh related stuff. (refreshMode would be enough, but the other ones do not make much sense either). 

For the HVAC stuff it will still be needed because my objective is that every temperature change will resend a command to the car if HVAC is on, so we will not be able to have a rule that applies for every single case...

**Comment by mschmiedel** [comment]
One more thought:

having the refreshMode and previousRefreshMode feels a bit overkill right now. Why not use 2 variablees, the refreshMode with PERIODIC|OFF, and forceRefresh as boolean. The command from the topic can set the forceRefresh to "true" everytime it is sent, and every successfulRefresh will reset it to false, this way, the refreshMode can be set only when really necessary without other "strange" logic...

**Comment by mschmiedel** [comment]
Actual state gives me a better feeling. We will need to test it though.

**Comment by tisoft** [comment]
I had this running now for over a week and have no problems anymore.

@mschmiedel If you are fine with the changes, would you approve the MR?

**Review by mschmiedel — APPROVED** [review]

File: saic-java-mqtt-gateway/src/main/java/net/heberling/ismart/mqtt/VehicleState.java [hunk 1]
```diff
@@ -623,9 +644,11 @@ public void setRefreshMode(RefreshMode refreshMode) {
           throw new MqttGatewayException("Error publishing message: " + mqttMessage, e);
         }
       }
+      this.previousRefreshMode = this.refreshMode;
+      this.refreshMode = refreshMode;
```
[the remark this answers is not in the document]
Comment (tisoft) [reply saic-java-mqtt-gateway/src/main/java/net/heberling/ismart/mqtt/VehicleState.java:648]: This seems to work for me, but we can surely improve it. :)

**Pull request merged**